backend/src/bundles/auth/auth.service.ts

```diff
@@ -11,10 +20,59 @@ class AuthService {
         this.userService = userService;
     }
 
-    public signUp(
+    public async signUp(
         userRequestDto: UserSignUpRequestDto,
     ): Promise<UserSignUpResponseDto> {
-        return this.userService.create(userRequestDto);
+        const foundUserByEmail = await this.userService.findByEmail(
+            userRequestDto.email,
+        );
+        if (foundUserByEmail) {
+            throw new HttpError({
+                message: ExceptionMessage.EMAIL_TAKEN,
+                status: HttpCode.BAD_REQUEST,
+            });
+        }
+        const { id } = await this.userService.create(userRequestDto);
+
+        const user = await this.userService.getUserWithProfile(id);
+
+        return {
+            accessToken: generateToken({ id }),
```

Comment:
For what purpose are you going to use the access token here?

Reply:
Some sites can give you access from the `sign-up` page immediately, for some user's it's kinda annoying to repeat the password after registration (I'm one of those), however, I've changed this as per the requirements :)